test/AspNet.Security.OAuth.Providers.Tests/Shopify/ShopifyTests.cs

```diff
@@ -0,0 +1,99 @@
+﻿/*
+ * Licensed under the Apache License, Version 2.0 (http://www.apache.org/licenses/LICENSE-2.0)
+ * See https://github.com/aspnet-contrib/AspNet.Security.OAuth.Providers
+ * for more information concerning the license and the contributors participating to this project.
+ */
+
+using System.Collections.Generic;
+using System.Net.Http;
+using System.Security.Claims;
+using System.Text;
+using System.Threading.Tasks;
+using AspNet.Security.OAuth.Infrastructure;
+using Microsoft.AspNetCore.Authentication;
+using Microsoft.AspNetCore.Builder;
+using Microsoft.Extensions.DependencyInjection;
+using Xunit;
+using Xunit.Abstractions;
+
+namespace AspNet.Security.OAuth.Shopify
+{
+    public class ShopifyTests : OAuthTests<ShopifyAuthenticationOptions>
+    {
+        private const string testShopName = "apple";
+
+        public ShopifyTests(ITestOutputHelper outputHelper)
+        {
+            OutputHelper = outputHelper;
+
+            LoopbackRedirectHandler = new ShopifyLoopbackRedirectHandler()
+            {
+                RedirectUri = RedirectUri,
+                ShopName = testShopName
+            };
+        }
+
+        public override string DefaultScheme => ShopifyAuthenticationDefaults.AuthenticationScheme;
+
+
+        protected internal override void ConfigureApplication(IApplicationBuilder app)
+        {
+            base.ConfigureApplication(app);
+
+            app.UseAuthentication();
+
+            app.Map("/me", childApp => childApp.Run(
```

Comment:
You could refactor the base class to allow the authentication properties to be provided some how, rather than having to copy-paste most of the implementation.